client/components/LearnerViewDirective.js

```diff
@@ -553,9 +588,28 @@ tie.directive('learnerView', [function() {
             } else if (syntaxErrorIndex === null) {
               $scope.syntaxErrorString = '';
               $scope.syntaxErrorFound = false;
+
+              // Updating reinforcement bullets only if no syntax errors
+              reinforcement.clear();
+              for (var caseList in reinforcementObject.passedList) {
+                if (reinforcementObject.passedList[caseList]) {
```

Comment:
Ah, I see. Let's not do that -- congratulatoryFeedback is a special case, and is just meant to show a message at the end of the task (which is just a display thing, and no logic is required). However, for something complex like reinforcement data, it's better to just pass the object wholesale from the service and do the logic within the service.